File: beginersSelection/abc083B-someSums/main.go
package main

import (
	"fmt"
	"math"
)

func main() {
	// Difine the variables.
	var NMax, A, B, sum, answer int

	// Define sum of each digit
	sum = 0

	// Answer
	answer = 0

	// Enter the values.
	fmt.Scanf("%d %d %d", &NMax, &A, &B)

	// Algolithm
	for N := 1; N <= NMax; N++ {

		given := N
		// 10^4.10^3,10^2,10^1,10^0で割って各桁の数を求めて和を取る
		for i := 0; i <= 4; i++ {
			quotient := given / int(math.Pow10(4-i))
			sum += quotient
			given -= int(math.Pow10(4-i)) * quotient
		}
		if A <= sum && sum <= B {
			answer += N
		}
		sum = 0
	}

	// Output the answer.
	fmt.Println(answer)
}

File: beginersSelection/abc083B-someSums/main.py
# # Input the values.
N, A, B = map(int, input().split())

# Define the variables.
sumsum = 0

# # Algorithm
for n in range(1, N+1):
    sum = 0
    tmp = n
    for i in range(5):
        d = 10**(4-i)
        r = int(n/d)
        sum += r
        n -= r*10**(4-i)
        # print(d, r, sum, n)

    if A <= sum <= B:
        sumsum += tmp

    n -= 1

# # Output
print(sumsum)
